TestStorageMonitoring is flaky

`TestStorageMonitoring` seems to be flaky on Windows tests example: https://github.com/elastic/apm-server/actions/runs/24683049355/job/72185077036?pr=20885

from the logs https://github.com/elastic/apm-server/actions/runs/24683049355/job/72185077036?pr=20885#step:5:2610 , this might be a data race issue where the bulk indexing failed or hasnt completed before the assertion is executed

In https://github.com/elastic/apm-server/actions/runs/28389808853/job/84113692872?pr=21290
```
[483](https://github.com/elastic/apm-server/actions/runs/28389808853/job/84113692872?pr=21290#step:5:2484)
=== RUN   TestStorageMonitoring
    logger.go:146: 2026-06-29T17:17:21.288Z	DEBUG	sampling	finalizing local sampling reservoirs
    logger.go:146: 2026-06-29T17:17:21.289Z	DEBUG	sampling	reporting 1 events
    logger.go:146: 2026-06-29T17:17:21.289Z	DEBUG	sampling	reporting 1 events
    logger.go:146: 2026-06-29T17:17:21.289Z	DEBUG	sampling	reporting 1 events
    logger.go:146: 2026-06-29T17:17:21.289Z	DEBUG	sampling	reporting 1 events
    logger.go:146: 2026-06-29T17:17:21.290Z	DEBUG	sampling	reporting 1 events
    logger.go:146: 2026-06-29T17:17:21.290Z	DEBUG	sampling	reporting 1 events
    logger.go:146: 2026-06-29T17:17:21.297Z	DEBUG	sampling	reporting 1 events
    logger.go:146: 2026-06-29T17:17:21.298Z	DEBUG	sampling	reporting 1 events
    logger.go:146: 2026-06-29T17:17:21.298Z	DEBUG	sampling	reporting 1 events
    logger.go:146: 2026-06-29T17:17:21.298Z	DEBUG	sampling	reporting 1 events
    logger.go:146: 2026-06-29T17:17:21.300Z	ERROR	v2@v2.14.1/appender.go:286	bulk indexing request failed	{"traceId": "00000000000000000000000000000000", "spanId": "0000000000000000", "error": "error decoding bulk response: ReadObjectCB: expect { or n, but found [, error found in #1 byte of ...|[{\"create\":|..., bigger context ...|[{\"create\":{\"_index\":\"\",\"_id\":\"\",\"_version\":0,\"resu|..."}
    main_test.go:129: 
        	Error Trace:	/Users/runner/work/apm-server/apm-server/x-pack/apm-server/main_test.go:129
        	Error:      	Not equal: 
        	            	expected: int(270000)
        	            	actual  : int64(0)
        	Test:       	TestStorageMonitoring
--- FAIL: TestStorageMonitoring (0.05s)
```

The bulk indexing is likely irrelevant, just noise from an imperfect ES mock.

--- a/x-pack/apm-server/sampling/processor_test.go
+++ b/x-pack/apm-server/sampling/processor_test.go
@@ -677,6 +677,8 @@ func TestStorageMonitoring(t *testing.T) {
 	})
 	require.NoError(t, err)
 	go processor.Run()
+	defer processor.Stop(context.Background())
+
 	for i := 0; i < 100; i++ {
 		traceID := uuid.Must(uuid.NewV4()).String()
 		batch := modelpb.Batch{{
@@ -693,11 +695,11 @@ func TestStorageMonitoring(t *testing.T) {
 		assert.Empty(t, batch)
 	}
 
-	// Stop the processor, flushing pending writes.
-	err = processor.Stop(context.Background())
-	require.NoError(t, err)
-
-	require.NoError(t, config.DB.Flush())
+	// Wait for cached db size update
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
+		lsm, _ := config.DB.Size()
+		assert.Greater(c, lsm, int64(0))
+	}, 2*time.Second, 20*time.Millisecond)
 
 	metricsNames := []string{
 		"apm-server.sampling.tail.storage.lsm_size",

